### PR TITLE
Creating Node Application Using Express Generator

### LINKED ISSUE #2290: Creating Node Application Using Express Generator
### Proposed title of article
 Creating Node Application Using Express Generator 

### Introduction paragraph (2-3 paragraphs):
 This article shows how you can create a "skeleton" website using the Express Application Generator tool, which you can then populate with site-specific routes, views/templates, and database calls. In this case, we'll use the tool to create the framework for our Local Library website, to which we'll later add all the other code needed by the site. The process is extremely simple, requiring only that you invoke the generator on the command line with a new project name, optionally also specifying the site's template engine and CSS generator.

### Key takeaways: 
1. Using the application generator
2. Creating the project using Express
3. Running the skeleton website build using Express

### PR DESCRIPTION
**Pull request by owinowendy**
### Article checklist

**ATTENTION: In your PR - add the ISSUE (topic form) # that this PR corresponds to. You can do this by also adding a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.**
      
      
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**If you have not already please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve our overall technical writing.**

#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**

this fixes #2290

### CODE REVIEW

File: articles/node-app-express-generator/index.md [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+
+```bash
+npx express-generator --view=pug expressExample
+
+```
+
+In the above command, we instruct the express-generator to generate an application called `expressExample`. We then specify the view that we need our application to use, in our case we select `pug`. There are many other view template engines that you could find [here](https://expressjs.com/en/guide/using-template-engines.html)
+
+Output:
+
```
Comment (paulodhiambo) [inline articles/node-app-express-generator/index.md:55]: Add the output here

**Review by paulodhiambo — CHANGES_REQUESTED** [review]
@owinowendy amazing article. Add the missing output section.

**Comment by paulodhiambo** [comment]
@owinowendy amazing article. @hectorkambow you can give it a final review.

**Comment by hectorkambow** [comment]
@owinowendy question why was this PR closed on the 16th?

**Comment by hectorkambow** [comment]
@paulodhiambo also - PRs need a plagiarism check before they can be ready for final.

**Comment by hectorkambow** [comment]
@owinowendy FYI PRs can only be created (and submitted) once the articles are completed. 

Please refrain from creating a PR to close and then re-open. Thank you. 

Cc @paulodhiambo

**Comment by hectorkambow** [comment]
@owinowendy 
What TOPIC form is this PR tied to???

**Comment by hectorkambow** [comment]
plagiarism check done

**Comment by hectorkambow** [comment]
@owinowendy 

Question just wanted to make sure that your tutorial does not overlap with this - it seems much of what is covered in your article may be published already??

https://www.section.io/engineering-education/express/

@paulodhiambo

**Comment by owinowendy** [comment]
The express package is the framework that exposes functionalities you can use in your code
The express-generator package a utility that provides a command-line tool you can use to scaffold your project - ie create boilerplate folder structure, files and code.

As part of the boiler plate files is a package.json file defining the dependencies for your project - ie npm packages that you will need for your project. The express package is listed there. This is where my article falls in.

If you want to build your project from scratch without boilerplate files / folders, first npm init, then npm install --save express, this installs express locally to your app, the --save option adds express as a dependency in your package.json.

Bottom line, to use the express web framework, you don't have to install express-generator, but you must install express. And if you work on, say, 3 apps which use express, the easiest thing to do is to install express 3 times locally to each app. This is where the published article falls in.    

The difference between this article and the one already published lies in the use of `Express Generator`  which is not covered in the existing article.

**Comment by hectorkambow** [comment]
@owinowendy 
Thank you - 👍 

Future article would benefit from that break down you provided me at the end - for a beginner audience - and it may be helpful to reference past articles if they talk about somewhat similar things 🚀
